src/Artsy/Analytics/trackingMiddleware.ts

```diff
@@ -43,13 +44,34 @@ export function trackingMiddleware(options: TrackingMiddlewareOptions = {}) {
           if (!foundExcludedPath) {
             logger.warn("Tracking PageView:", pathname)
 
-            analytics.page(
-              { path: pathname, referrer },
-              { integrations: { Marketo: false } }
-            )
+            const trackingData: {
+              path: string
+              referrer?: string
+              url: string
+            } = {
+              path: pathname,
+              url: sd.APP_URL + pathname,
+            }
+
+            // TODO: Remove after EXPERIMENTAL_APP_SHELL AB test ends.
+            if (getENV("EXPERIMENTAL_APP_SHELL")) {
+              trackingData.referrer = sd.APP_URL + referrer
+            }
+
+            // TODO: Remove after EXPERIMENTAL_APP_SHELL AB test ends.
+            if (
+              ["/collect", "/collections", "/collection/"].some(path =>
+                pathname.includes(path)
+              ) &&
+              referrer
```

Comment:
What's `referrer` here? I don't fully understand this.

Reply:
Oh nevermind!

Reply:
referrer means the previous path. This too is inferrered from segment if not explicitly provided. Since these pages are already client-side routing enabled, we check to see if there's already a referer in the router state on the client. This means we've navigated from say collection -> collection. Use that if so. If not, we've performed an SSR render (first init) so use the referrer from the server.